posixlib/src/main/scala/scala/scalanative/posix/signal.scala

```diff
@@ -0,0 +1,411 @@
+  @name("scalanative_sigusr1")
+  def SIGUSR1: CInt = extern
+  @name("scalanative_sigusr2")
+  def SIGUSR2: CInt = extern
+  // Note 1: macOS
+  // @name("scalanative_sigpoll")
+  // def SIGPOLL: CInt = extern
+  @name("scalanative_sigprof")
+  def SIGPROF: CInt = extern
+  @name("scalanative_sigsys")
+  def SIGSYS: CInt = extern
+  @name("scalanative_sigtrap")
+  def SIGTRAP: CInt = extern
+  @name("scalanative_sigurg")
+  def SIGURG: CInt = extern
+  @name("scalanative_sigtalrm")
+  def SIGVTALRM: CInt = extern
+  @name("scalanative_sigxcpu")
+  def SIGXCPU: CInt = extern
+  @name("scalanative_sigxfsz")
+  def SIGXFSZ: CInt = extern
+
+  // The storage occupied by sa_handler and sa_sigaction may overlap,
+  // and a conforming application shall not use both simultaneously.
+  type sigaction = CStruct4[
+    CFuncPtr1[CInt, Unit], // sa_handler Ptr to a signal-catching function or one of the SIG_IGN or SIG_DFL
+    sigset_t,              // sa_mask Set of signals to be blocked during execution of the signal handling func
+    CInt,                  // sa_flags Special flags
+    CFuncPtr3[CInt,
+              Ptr[siginfo_t],
+              Ptr[Byte],
+              Unit] // sa_sigaction Pointer to a signal-catching function
+  ]
+
+  // define the following macros which shall expand to integer constant expressions
+  // that need not be usable in #if preprocessing directives
+  @name("scalanative_sig_block")
+  def SIG_BLOCK: CInt = extern
+  @name("scalanative_sig_unblock")
+  def SIG_UNBLOCK: CInt = extern
+  @name("scalanative_sig_setmask")
+  def SIG_SETMASK: CInt = extern
+
+  // define the following symbolic constants
+  @name("scalanative_sa_nocldstop")
+  def SA_NOCLDSTOP: CInt = extern
+  @name("scalanative_sa_onstack")
+  def SA_ONSTACK: CInt = extern
+  @name("scalanative_sa_resethand")
+  def SA_RESETHAND: CInt = extern
+  @name("scalanative_sa_restart")
+  def SA_RESTART: CInt = extern
+  @name("scalanative_sa_siginfo")
+  def SA_SIGINFO: CInt = extern
+  @name("scalanative_sa_nocldwait")
+  def SA_NOCLDWAIT: CInt = extern
+  @name("scalanative_sa_nodefer")
+  def SA_NODEFER: CInt = extern
+  @name("scalanative_ss_onstack")
+  def SS_ONSTACK: CInt = extern
+  @name("scalanative_ss_disable")
+  def SS_DISABLE: CInt = extern
+  @name("scalanative_minsigstksz")
+  def MINSIGSTKSZ: CInt = extern
+  @name("scalanative_sigstksz")
+  def SIGSTKSZ: CInt = extern
+
+  // A machine-specific representation of the saved context
+  // mac OS type mcontext_t = Ptr[__darwin_mcontext64]
+  // __darwin_mcontext64 -> _STRUCT_MCONTEXT64 -> typedef _STRUCT_MCONTEXT64	*mcontext_t;
+  type mcontext_t = Ptr[Byte]
+
+  type ucontext_t = CStruct4[
+    Ptr[Byte], // ucontext_t *uc_link Ptr to the context that is resumed when this context returns (Ptr instead)
+    sigset_t,  // c_sigmask The set of signals that are blocked when this context is active
+    Ptr[
+      stack_t
+    ],         // uc_stack The stack used by this context (Ptr instead of value)
```

Comment:
Was such formatting done by scalafmt? Looks a bit weird. 
Anyway, I think it looks good overall. Can you please rebase it onto the current master to make sure that it passes check-lint in the CI?

Reply:
It has to do with the alignment being `most` trying to align thing vertically and then the comment was too long. I shortened the comment enough to make it work for now. We should discuss trying to get scalafmt upgraded at some point but we will need to change some settings and accept some changes.

Reply:
I guess I missed this one on the prior review, sorry.